ipaserver/plugins/group.py

```diff
@@ -52,7 +52,7 @@
 
 logger = logging.getLogger(__name__)
 
-if api.env.in_server and api.env.context in ['lite', 'server']:
+if api.env.in_server and api.env.context in ['lite', 'server', 'cli']:
```

Comment:
A subtle but perhaps important change. Given that the plugins no longer reside on clients maybe this is ok.

I wonder if a different way would be to allow the context to be set via ipa -e so that one could do:

```
ipa -e context=server console
```

It currently doesn't work. The context is forced to cli.

Reply:
I'd rather add 'cli' support because it is going to ease our life in future for other uses around fixing trust configurations or similar stuff.

Reply:
I just wonder about unexpected or unwanted side-effects of allowing cli to execute server-side-only code. In the end forcing them to specify cli as the context may be splitting hairs but at least they'll knowingly be doing that.

Reply:
Well, I consider it a bug to not allow `ipa console` in server mode to operate the same way as under `httpd`. May be we should instead fix context to be `server` when `ipa -e in_server=True console` is run?

Reply:
The context checking confuses me. The only place this is done is on some plugins, the framework relies only on in_server to determine its actions. Did this perhaps pre-date the lite client? I suspect that in_server is sufficient. For the sake of time and it seems like a no-op I won't object to adding cli for the purposes of least harm but I'd want a ticket to investigate this further.